Formula/php.rb

```diff
@@ -0,0 +1,324 @@
+      --enable-calendar
+      --enable-dba
+      --enable-dtrace
+      --enable-exif
+      --enable-ftp
+      --enable-fpm
+      --enable-gd-native-ttf
+      --enable-intl
+      --enable-mbregex
+      --enable-mbstring
+      --enable-mysqlnd
+      --enable-pcntl
+      --enable-phpdbg
+      --enable-phpdbg-webhelper
+      --enable-shmop
+      --enable-soap
+      --enable-sockets
+      --enable-sysvmsg
+      --enable-sysvsem
+      --enable-sysvshm
+      --enable-wddx
+      --enable-zip
+      --libexecdir=#{libexec}
+      --with-apxs2=#{Formula["httpd"].opt_bin}/apxs
+      --with-bz2
+      --with-enchant=#{Formula["enchant"].opt_prefix}
+      --with-fpm-user=_www
+      --with-fpm-group=_www
+      --with-freetype-dir=#{Formula["freetype"].opt_prefix}
+      --with-gd
+      --with-gettext=#{Formula["gettext"].opt_prefix}
+      --with-gmp=#{Formula["gmp"].opt_prefix}
+      --with-iconv-dir
+      --with-icu-dir=#{Formula["icu4c"].opt_prefix}
+      --with-imap=shared,#{Formula["imap-uw"].opt_prefix}
+      --with-imap-ssl=#{Formula["openssl"].opt_prefix}
+      --with-jpeg-dir=#{Formula["jpeg"].opt_prefix}
+      --with-kerberos
+      --with-ldap=shared
+      --with-ldap-sasl
+      --with-libedit
+      --with-libxml-dir
+      --with-libzip=#{Formula["libzip"].opt_prefix}
+      --with-mhash
+      --with-mysql-sock=/tmp/mysql.sock
+      --with-mysqli=mysqlnd
+      --with-ndbm
+      --with-openssl=#{Formula["openssl"].opt_prefix}
+      --with-pdo-dblib=#{Formula["freetds"].opt_prefix}
+      --with-pdo-mysql=mysqlnd
+      --with-pdo-odbc=unixODBC,#{Formula["unixodbc"].opt_prefix}
+      --with-pdo-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-pic
+      --with-png-dir=#{Formula["libpng"].opt_prefix}
+      --with-pspell=#{Formula["aspell"].opt_prefix}
+      --with-snmp
+      --with-tidy=shared,#{Formula["tidy-html5"].opt_prefix}
+      --with-unixODBC=#{Formula["unixodbc"].opt_prefix}
+      --with-webp-dir=#{Formula["webp"].opt_prefix}
+      --with-xmlrpc
+      --with-zlib
+      --with-xsl
+    ]
+
+    if MacOS.version < :lion
+      args << "--with-curl=#{Formula["curl"].opt_prefix}"
+    else
+      args << "--with-curl"
+    end
+
+    if build.devel?
+      args += %W[
+        --with-password-argon2=#{Formula["argon2"].opt_prefix}
+        --with-sodium=#{Formula["libsodium"].opt_prefix}
+      ]
+    else
+      args << "--with-mcrypt=shared,#{Formula["mcrypt"].opt_prefix}"
+    end
+
+    system "./configure", *args
+
+    inreplace "Makefile" do |s|
+      # Custom location for php module and remove -a (don't touch httpd.conf)
+      s.gsub! /^INSTALL_IT = \$\(mkinstalldirs\) '([^']+)' (.+) LIBEXECDIR=([^\s]+) (.+) -a (.+)$/,
+        "INSTALL_IT = $(mkinstalldirs) '#{lib}/httpd/modules' \\2 LIBEXECDIR='#{lib}/httpd/modules' \\4 \\5"
+
+      # Reorder linker flags to put system paths at the end to avoid accidential system linkage
```

Comment:
What does this mean? If accidental system linkage is possible for a given library, we should use the system version anyway, and drop the dependency.

Reply:
We run into situations where the system library is the wrong version (tidy) or is missing the necessary header files (net-snmp, libpq).

Reply:
If it's the wrong version of tidy, why is the build system willing to link to it? We shouldn't be having to patch the Makefile to avoid such a thing from happening.

Reply:
The build system recognies the configure flag for the Homebrew tidy version and uses those headers, but because the system folder is added, by the build system, to the linker first (for other deps) that folder gets preference over the additional Homebrew folder that comes later in the flag.

Reply:
They need to fix that.

Reply:
I'll do my best to report my findings upstream, I just don't know if autoconf can handle this particular edge case.

The issue is moot for tidy if we build it as shared after the main binaries with `phpize`. I imagine we could try to do the same with the other extensions that have system collisions.

Reply:
The Makefile is currently generated like this 
```
EXTRA_LDFLAGS = -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/enchant/lib -L/usr/local/opt/webp/lib -L/usr/local/opt/jpeg/lib -L/usr/local/opt/libpng/lib -L/usr/local/opt/freetype/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/gmp/lib -L/usr/local/Cellar/icu4c/59.1/lib -L/usr/local/opt/unixodbc/lib -L/usr/local/opt/freetds/lib -L/usr/local/Cellar/libpq/9.6.5/lib -L/usr/local/opt/aspell/lib -L/usr/local/Cellar/net-snmp/5.7.3/lib -L/usr/local/opt/libzip/lib
EXTRA_LDFLAGS_PROGRAM = -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/enchant/lib -L/usr/local/opt/webp/lib -L/usr/local/opt/jpeg/lib -L/usr/local/opt/libpng/lib -L/usr/local/opt/freetype/lib -L/usr/local/opt/gettext/lib -L/usr/local/opt/gmp/lib -L/usr/local/Cellar/icu4c/59.1/lib -L/usr/local/opt/unixodbc/lib -L/usr/local/opt/freetds/lib -L/usr/local/Cellar/libpq/9.6.5/lib -L/usr/local/opt/aspell/lib -L/usr/local/Cellar/net-snmp/5.7.3/lib -L/usr/local/opt/libzip/lib
```
which you say causes the issue since the system path appears first. Is that correct @kabel ?

Reply:
A similar issue has been reported here https://bugs.php.net/bug.php?id=68313

Reply:
All looks correct to me. Good find on the open upstream bug.